File: unificacion_maga/add_drop_tables.py
import re

sql_file = r'c:\xampp\htdocs\unificacion_maga\Databases\vider_maga.sql'

with open(sql_file, 'r', encoding='utf-8') as f:
    content = f.read()

# Buscamos patrones como: CREATE TABLE `nombre` (
# y lo reemplazamos por:
# DROP TABLE IF EXISTS `nombre`;
# CREATE TABLE `nombre` (

pattern = r'CREATE TABLE `([^`]+)` \('

def replacer(match):
    table_name = match.group(1)
    return f'DROP TABLE IF EXISTS `{table_name}`;\nCREATE TABLE `{table_name}` ('

new_content = re.sub(pattern, replacer, content)

with open(sql_file, 'w', encoding='utf-8') as f:
    f.write(new_content)

print("Archivo SQL actualizado con DROP TABLE IF EXISTS.")

File: unificacion_maga/add_drop_tables.php
<?php
$sql_file = __DIR__ . '/Databases/vider_maga.sql';

$content = file_get_contents($sql_file);

$pattern = '/CREATE TABLE `([^`]+)` \(/';
$replacement = "DROP TABLE IF EXISTS `$1`;\nCREATE TABLE `$1` (";

$new_content = preg_replace($pattern, $replacement, $content);

file_put_contents($sql_file, $new_content);

echo "Archivo SQL actualizado con DROP TABLE IF EXISTS.";
